StyleCop.Analyzers/StyleCop.Analyzers/DocumentationRules/SA1600ElementsMustBeDocumented.cs

```diff
@@ -44,7 +50,195 @@ public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics
+
+            if (declaration != null && NeedsComment(declaration.Modifiers, isDefinedInClassOrStruct ? SyntaxKind.PrivateKeyword : SyntaxKind.PublicKeyword))
+            {
+                var leadingTrivia = declaration.GetLeadingTrivia();
+                var commentTrivia = leadingTrivia.FirstOrDefault(x => x.IsKind(SyntaxKind.SingleLineDocumentationCommentTrivia));
+                if (XmlCommentHelper.IsMissingOrEmpty(commentTrivia))
+                {
+                    context.ReportDiagnostic(Diagnostic.Create(Descriptor, declaration.Identifier.GetLocation()));
+                }
+            }
+        }
+
+        private void HandleEventDeclaration(SyntaxNodeAnalysisContext context)
+        {
+            EventDeclarationSyntax declaration = context.Node as EventDeclarationSyntax;
+
+            if (declaration != null && NeedsComment(declaration.Modifiers, SyntaxKind.PrivateKeyword))
+            {
+                var leadingTrivia = declaration.GetLeadingTrivia();
+                var commentTrivia = leadingTrivia.FirstOrDefault(x => x.IsKind(SyntaxKind.SingleLineDocumentationCommentTrivia));
+                if (XmlCommentHelper.IsMissingOrEmpty(commentTrivia))
+                {
+                    context.ReportDiagnostic(Diagnostic.Create(Descriptor, declaration.Identifier.GetLocation()));
+                }
+            }
+        }
+
+        private void HandleEventFieldDeclaration(SyntaxNodeAnalysisContext context)
+        {
+            EventFieldDeclarationSyntax declaration = context.Node as EventFieldDeclarationSyntax;
+            var variableDeclaration = declaration?.Declaration;
+
+            if (variableDeclaration != null && NeedsComment(declaration.Modifiers, SyntaxKind.PrivateKeyword))
+            {
+                var leadingTrivia = declaration.GetLeadingTrivia();
+                var commentTrivia = leadingTrivia.FirstOrDefault(x => x.IsKind(SyntaxKind.SingleLineDocumentationCommentTrivia));
+                if (XmlCommentHelper.IsMissingOrEmpty(commentTrivia))
+                {
+                    var locations = variableDeclaration.Variables.Select(v => v.Identifier.GetLocation()).ToArray();
+                    if (locations.Length > 0)
+                    {
+                        context.ReportDiagnostic(Diagnostic.Create(Descriptor, locations.First()));
+                    }
+                }
+            }
+        }
+
+        private bool NeedsComment(SyntaxTokenList modifiers, SyntaxKind defaultModifier)
+        {
+
+            return (modifiers.Any(SyntaxKind.PublicKeyword)
+                || modifiers.Any(SyntaxKind.ProtectedKeyword)
+                || modifiers.Any(SyntaxKind.InternalKeyword)
+                || defaultModifier == SyntaxKind.PublicKeyword
+                || defaultModifier == SyntaxKind.ProtectedKeyword
+                || defaultModifier == SyntaxKind.InternalKeyword)
```

Comment:
If we follow StyleCops implementation then it should be reported for internal members